account_default_draft_move/invoice_view.xml

```diff
@@ -22,7 +22,9 @@
       <field name="type">form</field>
       <field name="arch" type="xml">
         <xpath expr="//button[@name='invoice_cancel']" position="after">
-          <button name="invoice_cancel" states="open" string="Cancel Invoice" groups="account.group_account_invoice"/>
+          <field name="update_posted" invisible="1"/>
+          <button name="invoice_cancel" states="open" string="Cancel Invoice" groups="account.group_account_invoice"
+                  attrs="{'invisible': ['|', ('update_posted', '=', False)]}"/>
```

Comment:
@sbidoul there is one cancel button in the form (https://github.com/odoo/odoo/blob/8.0/addons/account/account_invoice_view.xml#L150)
and a second one in this module.

If you use states and invisible, you should take account of state in invisible attribute.
In this module, the cancel button should be visible only if update_posted is true and invoice is open (in the other case, the original button is displayed)
So invisible if not update_posted OR not open.